### PR TITLE
Cache test data on GHA

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by maxulysse** [comment]
Should be able to delete https://github.com/nf-core/test-datasets/blob/sarek3/tests/config/test_data.config and use the usual file from modules, which is more future proof, once and if https://github.com/nf-core/modules/pull/2739 is merged

**Review by nvnieuwk — APPROVED** [review]
Looks awesome! I'm not a CI expert though, so it may be better to wait for a second review

File: nextflow_schema.json
Comment (pinin4fjords) [inline nextflow_schema.json:795]: Maybe a bit better description on why you might use this option? I know it's mostly for the CI, but still, users will see this.

**Review by pinin4fjords — APPROVED** [review]
Looks snazzy! FWIW I'd have reviewed quicker if there had a been a description ;-)

File: nextflow_schema.json
Comment (maxulysse) [reply nextflow_schema.json:795]: It's an hidden params with a default value. I think it should be fine

**Comment by maxulysse** [comment]
> Looks snazzy! FWIW I'd have reviewed quicker if there had a been a description ;-)

Good point. I'll pay more attention to that

**Pull request merged**
